### PR TITLE
create class functionality for frogger game

### PR DESCRIPTION
**Pull request by yulyasystem**
### task name

[Demo](https://yulyasystem.github.io/frogger-game/) |
[Code base](https://github.com/yulyasystem/frogger-game/blob/master/js/app.js)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/yulyasystem/frogger-game/js/app.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  Enemy.prototype.update = function(dt) {
+    this.x += this.speed * dt;
+    this.handleCollision();
+  };
+  
+  Enemy.prototype.handleCollision = function() {
+    if (this.x > block.x * 5) {
+      this.x = -block.x;
+      this.speed = generateSpeed();
+    }
```
Comment (OleksiyRudenko) [inline submissions/yulyasystem/frogger-game/js/app.js]: `handleCollision` is expected to handle collisions. But it seems it does something else. Resets character position under certain conditions?

Please, either explain or fix.

The good practice is to follow the Separation of Concerns principle, which requires entities like functions have a single major responsibility.

File: submissions/yulyasystem/frogger-game/js/app.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        this.x > 1 ? (this.x = this.x - block.x) : null;
+        break;
+      case "right":
+        this.x < block.x * 4 ? (this.x = this.x + block.x) : null;
+        break;
+    }
+  };
+
+  const player = new Player(initial.x, initial.y);
+  let allEnemies =[block.y, block.y*2, block.y*3].map(location => new Enemy(-block.x, location, generateSpeed()));
```
Comment (OleksiyRudenko) [inline submissions/yulyasystem/frogger-game/js/app.js]: Not required to fix but keep in mind that long strings can be split at chaining points (`.`) for better readability.
Example:
```
let allEnemies = [block.y, block.y*2, block.y*3]
  .map(location => new Enemy(-block.x, location, generateSpeed()));
```

File: submissions/yulyasystem/frogger-game/js/app.js [hunk 3]
```diff
@@ -1,97 +1,97 @@
...
+      break;
+    case "right":
+      this.x < block.x * 4 ? (this.x = this.x + block.x) : null;
+      break;
+  }
+};
+
+const player = new Player(initial.x, initial.y);
+let allEnemies = [block.y, block.y * 2, block.y * 3]
+.map(location => new Enemy(-block.x, location, generateSpeed()));
```
Comment (OleksiyRudenko) [inline submissions/yulyasystem/frogger-game/js/app.js]: Thank you fixing formatting.
Chaining or expression continuation follows indentation increase rules.

Not required to fix this now. Just keep in mind.
Your IDE may support code linters and/or prettifier. Please, google for an appropriate plug-in
It makes code well formatted at no effort. You don't need spending much of your time for this.

Contextual formatting examples: 
```
let whatever = [any]
  .map()
  .join()
  .fetch()
  .then()
  .catch();

let playList = [
    'Bohemian Rhapsody',
    'Zombie',
    'Paint It Black',
  ];

for (let i = 0; i < 100; i++) {
  i--;
}
```

You may set your IDE

Comment (yulyasystem) [reply submissions/yulyasystem/frogger-game/js/app.js]: Thank you!

**Pull request merged**